### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by batoolfatima2135**
Hi, 👋
In this branch I have :
- Added RSpec as a gem to project.
- Created a Request spec file for all controllers.
- For each action checked:
    - If response status was correct.
    - If a correct template was rendered.
    - If the response body includes correct placeholder text.

### CODE REVIEW

File: spec/requests/posts_spec.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  end
+
+  describe 'GET /index' do
+    before :each do
+      @user = User.create! valid_attributes
+      get user_posts_path(@user)
+    end
+    it 'returns a successful response' do
+      expect(response).to be_successful
+    end
```
Comment (Abenezer-Tilahun) [inline spec/requests/posts_spec.rb:17]: - [ ] Good job for adding each action you check: If response status was correct, If a correct template was rendered and If the response body includes correct placeholder text But Kindly makes sure that all test cases are passing when I run on my terminal I got one of your test case fail So kindly Fix it.
test


test 1

**Review by Abenezer-Tilahun — CHANGES_REQUESTED** [review]
Hi @batoolfatima2135 

Great job for implemented  the some of the Project Requirement  , but there are some changes that need to be implemented in order to proceed to the next project 👍

## Highlights 👏🏻
✔️ No Linter errors
✔️ Correct Git-Flow
✔️ Good commit messages
✔️ Professional README
✔️ Correctly Setup and configure ` RSpec `as a gem to your project..
✔️ Correctly Create a Request spec file for all your controllers.


almost

## Required Changes ♻️
_Check the comments below the review._

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_spec.rb (hunk 1 above)
Comment (batoolfatima2135) [reply spec/requests/posts_spec.rb:17]: Hi @Abenezer-Tilahun , on my end every test is passing!
Capture
What should I do now?? 🤔

**Review by frank1738 — APPROVED** [review]
Hi @batoolfatima2135 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

✅ Good work setting `rspec`
✅ All tests are passing

image


## Optional suggestions

N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**